harpy/downsample.py

```
@@ -0,0 +1,191 @@
+        re_ext = re.compile(r"\.(fq|fastq)(?:\.gz)?$", re.IGNORECASE)
+        badfastq = []
+        for i in input:
+            if not re.search(re_ext, i):
+                badfastq.append(i)
+        if badfastq:
+            raise click.BadParameter('inputs must be 1 BAM (.bam) file or 2 FASTQ (.fastq|.fq) files. The FASTQ files can be gzipped.')
+
+    output_bam = f"{prefix}.bam"
+    bx_tag = bx_tag.upper()
+    rng = random.Random(random_seed).sample if random_seed else random.Random().sample
+    logfile = open(f"{prefix}.log", "w")
```

Ruff (0.8.0)

Use a context manager for opening files

(SIM115)